src/execution/BucketExecutor.tsx

```diff
@@ -0,0 +1,338 @@
+            if (nextProps[propName] !== prevState[propName]) {
+                newState[propName] = nextProps[propName];
+            }
+        });
+        if (Object.keys(newState).length === 0) {
+            return null;
+        }
+
+        newState.isLoading = nextProps.autoLoadFirstPage;
+        newState.response = null;
+        newState.responsePromise = null;
+        newState.result = null;
+        newState.resultPromise = null;
+
+        return newState;
+    }
+
+    public componentDidUpdate(prevProps: IBucketExecutorProps, prevState: IBucketExecutorState) {
+        if (this.props.sdk !== prevProps.sdk) {
+            // tslint:disable-next-line:no-console
+            console.warn(
+                "Do not change the sdk prop once set. If you need to do this, force remount using the key attribute.",
+            );
+        }
+        if (didPropsChange(propsRequiringDataReset, prevState, this.state) && this.props.autoLoadFirstPage) {
+            this.getResponse();
+        }
+    }
+
+    public state: IBucketExecutorState;
+
+    public hasUnmounted: boolean = false;
+
+    private sdk: SDK;
+
+    constructor(props: IBucketExecutorProps) {
+        super(props);
+        this.state = {
+            result: null,
+            response: null,
+            isLoading: props.autoLoadFirstPage,
+            error: null,
+            responsePromise: null,
+            resultPromise: null,
+            children: props.children,
+            projectId: props.projectId,
+            dimensions: props.dimensions,
+            filters: props.filters,
+            sortBy: props.sortBy,
+        };
+        this.sdk = setupSDK(props);
+
+        this.hasUnmounted = false;
+    }
+
+    public componentDidMount() {
+        if (this.props.autoLoadFirstPage) {
+            this.getResponse();
+        }
+    }
+
+    public componentWillUnmount() {
+        this.hasUnmounted = true;
+    }
+
+    public getPage = (paging: IPaging) => {
+        const { response } = this.state;
+        if (this.hasUnmounted) {
+            // tslint:disable-next-line:no-console
+            console.warn("You are calling a component that is already unmounted");
+        }
+        if (!response) {
+            this.getResponse(true, paging);
+        } else {
+            this.getResult(response, paging);
+        }
+    };
+
+    public getResponse = (
+        triggerGetResult: boolean = this.props.autoLoadFirstPage,
+        resultPaging = this.getInitialPaging(),
+    ) => {
+        const { dimensions, filters, sortBy, projectId } = this.state;
+        const execution = getExecutionFromDimensions(dimensions, filters, sortBy);
+        const responsePromise = this.sdk.execution.getExecutionResponse(projectId, execution);
+        this.setState({
+            result: null,
+            response: null,
+            isLoading: true,
+            error: null,
+            responsePromise,
+        });
+        responsePromise.then(
```

Comment:
Question: Is there a reason you chose this instead of `responsePromise.then(response => ...).catch(error => ...)`. I'm not saying this is somehow wrong, just that we don't use it much.